### PR TITLE
[Sprint: 41] XD-2643 Kafka test creates Topic prior to tests

### PR DESCRIPTION
**Pull request by markpollack**

### CODE REVIEW

**Comment by markpollack** [comment]
Need to put the code for making sure the topic (metadata) was created as done in the message bus.  Add to another utility method in KafkaUtils...

```java
				TopicMetadata topicMetadata = retryTemplate.execute(new RetryCallback<TopicMetadata, Exception>() {
					@Override
					public TopicMetadata doWithRetry(RetryContext context) throws Exception {
						TopicMetadata topicMetadata = AdminUtils.fetchTopicMetadataFromZk(topicName, zkClient);
						if (topicMetadata.errorCode() != ErrorMapping.NoError() || !topicName.equals(topicMetadata.topic())) {
							// downcast to Exception because that's what the error throws
							throw (Exception) ErrorMapping.exceptionFor(topicMetadata.errorCode());
						}
						List<PartitionMetadata> partitionMetadatas = new kafka.javaapi.TopicMetadata(topicMetadata).partitionsMetadata();
						for (PartitionMetadata partitionMetadata : partitionMetadatas) {
							if (partitionMetadata.errorCode() != ErrorMapping.NoError()) {
								throw (Error)ErrorMapping.exceptionFor(partitionMetadata.errorCode());
							}
						}
						return topicMetadata;
					}
				});
```

File: spring-xd-test-fixtures/src/main/java/org/springframework/xd/test/fixtures/KafkaSink.java [hunk 1]
```diff
@@ -54,18 +54,18 @@ public static KafkaSink withDefaults() {
...
 	 *
-	 * @param brokerList A list of brokers that the sink can connect
+	 * @param zkConnect A list of brokers that the sink can connect
 	 * @return instance of Kafka sink
 	 */
-	public KafkaSink brokerList(String brokerList) {
-		Assert.hasText(brokerList, "brokerList must not be empty nor null");
-		this.brokerList = brokerList;
+	public KafkaSink brokerList(String zkConnect) {
+		Assert.hasText(zkConnect, "brokerList must not be empty nor null");
```
Comment (markpollack) [inline spring-xd-test-fixtures/src/main/java/org/springframework/xd/test/fixtures/KafkaSink.java]: I forgot to update 'brokerList' to zkConnect  ... should be consistent everywhere....

File: spring-xd-test-fixtures/src/main/java/org/springframework/xd/test/fixtures/util/KafkaUtils.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+/**
+ * Utility methods for working with Kafka.  Used by {@link org.springframework.xd.test.fixtures.KafkaSource} and
+ * {@link org.springframework.xd.test.fixtures.KafkaSink}
+ *
+ * @author Mark Pollack
+ */
+public class KafkaUtils {
+
+    /**
+     * Ensures that the host and port of the zookeeper connection are available
```
Comment (ericbottard) [inline spring-xd-test-fixtures/src/main/java/org/springframework/xd/test/fixtures/util/KafkaUtils.java]: Should mention that it also creates the topic

File: spring-xd-test-fixtures/src/main/java/org/springframework/xd/test/fixtures/AvailableSocketPorts.java [hunk 3]
```diff
@@ -53,14 +53,14 @@ public static synchronized int nextAvailablePort() {
 	/**
 	 * Verifies that the port to the broker is available. If not throws an IllegalStateException.
```
Comment (ericbottard) [inline spring-xd-test-fixtures/src/main/java/org/springframework/xd/test/fixtures/AvailableSocketPorts.java]: I don't know if I'm really the original author of this, but this javadoc is really misleading as it's doing quite the opposite: it ensures that the port is *listening*

File: spring-xd-test-fixtures/src/main/java/org/springframework/xd/test/fixtures/util/KafkaUtils.java (hunk 2 above)
Comment (cppwfs) [inline spring-xd-test-fixtures/src/main/java/org/springframework/xd/test/fixtures/util/KafkaUtils.java]: Will fix on merge

File: spring-xd-test-fixtures/src/main/java/org/springframework/xd/test/fixtures/AvailableSocketPorts.java (hunk 3 above)
Comment (cppwfs) [inline spring-xd-test-fixtures/src/main/java/org/springframework/xd/test/fixtures/AvailableSocketPorts.java]: Will fix on merge

**Comment by cppwfs** [comment]
Fixed, Updated with code review comments, rebased, squashed merged.

**Pull request closed without merging**